.github/workflows/release.yml

```diff
@@ -1,15 +1,28 @@
 on:
+  # You must provide a tag either via the action input or the git ref (i.e push / create a tag).
+  # If you do not provide a tag the action will fail.
+
+  # If the tag of the release you are creating does not yet exist, you should set
+  # both the tag and commit action inputs. commit can point to a commit hash or a branch name (ex - main).
+#   workflow_dispatch:
+#     inputs:
+#       branch:
+#         description: 'Define branch name to run the release GH action against'
+#         required: true
+#         default: 'main'
   release:
-    types: [published]
+    type: [created]
```

Comment:
Why do you change from `published` to `created`? The documentation is very poor but if I understand correctly, `created` will trigger the github action when setting a draft new release. Whereas `published` will trigger the github action in the previous example or when publishing directly (apparently via some special scripts)

Reply:
`[published]` would require the release to already be created (either manually or via automation) and based on my other changes, switching to `[created]` allows a maintainer (with the proper perms) to create a release but not publish it. This ties into the github  actions configuration [here](https://github.com/flannel-io/cni-plugin/pull/22/files#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34R195-R206) as we will be able to update existing releases using our automation. 

The process would be:
- a maintainer creates a new release 
- github actions will kick off automatically
- tests will run and assuming that everything passes, the existing release will then have the release artifacts automatically attached to it and the corresponding docker images will also be published.

I have an example run [here](https://github.com/rosskirkpat/cni-plugin/runs/4621973411?check_suite_focus=true) and  a test release [here](https://github.com/rosskirkpat/cni-plugin/releases/tag/v1.0.0-rc7). The build and publish steps were expected to fail as I didn’t substitute in my own registry/creds

Reply:
ok thanks